Enable all disabled checks options in golangci-lint, see #2786

In #2786 we started to enable some of the disabled checks in golangci-lint, but not all of them are enabled because of the amount of work required.

This is a tracking issue to ensure all are enabled.

I have tried to renew the experiment from @mx-psi with #9133 and have the same results. Most of the code is not in the hot path.

The fieldalignment linter picked up interesting improvements on the protobuf code, however, and this might be something to pursue down the road in the opentelemetry-proto project.

I have reviewed and tried to apply this to different parts of the code, but I don't see a good target for improvement because none of the code is in the hot path. I identified that the generated protobuf code could use a boost. Unfortunately protobuf values the ordering of fields.

Traces:
```
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/trace/v1/trace.pb.go:179:20: struct with 88 pointer bytes could be 72
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/trace/v1/trace.pb.go:253:17: struct with 96 pointer bytes could be 80
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/trace/v1/trace.pb.go:321:11: struct of size 208 could be 192
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/trace/v1/trace.pb.go:525:17: struct with 32 pointer bytes could be 24
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/trace/v1/trace.pb.go:605:16: struct with 48 pointer bytes could be 24
```


Metrics:
```
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:219:22: struct with 88 pointer bytes could be 72
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:293:19: struct with 96 pointer bytes could be 80
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:445:13: struct with 64 pointer bytes could be 56
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:883:22: struct with 64 pointer bytes could be 48
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:1038:25: struct with 176 pointer bytes could be 128
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:1283:36: struct of size 216 could be 208
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:1561:44: struct with 16 pointer bytes could be 8
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:1627:23: struct with 64 pointer bytes could be 32
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go:1814:15: struct with 48 pointer bytes could be 24
```

Logs:
```
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/logs/v1/logs.pb.go:221:19: struct with 88 pointer bytes could be 72
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/logs/v1/logs.pb.go:295:16: struct with 96 pointer bytes could be 80
/Users/atoulme/w/opentelemetry-collector/pdata/internal/data/protogen/logs/v1/logs.pb.go:362:16: struct with 64 pointer bytes could be 40
```

In the meantime, I move that we remove the TODO in the .golangci-lint.yaml file and close the task to revisit the fieldalignment linter.

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -69,8 +69,11 @@ linters-settings:
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
 
     enable-all: true
-    # TODO: Enable this and fix the alignment issues.
     disable:
+      # We want to order fields according to readability and grouping them by use cases.
+      # This linter does not offer a discernible performance improvement as the structs
+      # defined in this repository are not in the execution hot path.
+      # See https://github.com/open-telemetry/opentelemetry-collector/issues/2789
       - fieldalignment
 
   revive:

